src/ripple/app/misc/HashRouter.cpp

```diff
@@ -26,124 +26,85 @@
 
 namespace ripple {
 
-std::function<bool(STTx const&, std::function<bool(STTx const&)>)>
-HashRouter::sigVerify()
+// These are the same flags defined as SF_PRIVATE1-4 in the header
+#define SF_SIGBAD      0x100    // Signature is bad
+#define SF_SIGGOOD     0x200    // Signature is good
+#define SF_LOCALBAD    0x400    // Local checks failed
+#define SF_LOCALGOOD   0x800    // Local checks passed
+
+auto
+HashRouter::emplace (uint256 const& index)
+    -> std::pair<Entry&, bool>
 {
-    return
-    [&] (STTx const& tx, std::function<bool(STTx const&)> sigCheck)
-    {
-        auto const id = tx.getTransactionID();
-        auto const flags = getFlags(id);
-        if (flags & SF_SIGGOOD)
-            return true;
-        if (flags & SF_BAD)
-            return false;
-        if (! sigCheck(tx))
-        {
-            setFlags(id, SF_BAD);
-            return false;
-        }
-        setFlags(id, SF_SIGGOOD);
-        return true;
-    };
-}
-
-HashRouter::Entry& HashRouter::findCreateEntry (uint256 const& index, bool& created)
-{
-    hash_map<uint256, Entry>::iterator fit = mSuppressionMap.find (index);
+    auto fit = mSuppressionMap.find (index);
 
     if (fit != mSuppressionMap.end ())
     {
-        created = false;
-        return fit->second;
+        return std::make_pair(
+            std::ref(fit->second), false);
     }
 
-    created = true;
-
-    int now = UptimeTimer::getInstance ().getElapsedSeconds ();
-    int expireTime = now - mHoldTime;
+    int elapsed = UptimeTimer::getInstance ().getElapsedSeconds ();
+    int expireCutoff = elapsed - mHoldTime;
 
     // See if any supressions need to be expired
-    std::map< int, std::list<uint256> >::iterator it = mSuppressionTimes.begin ();
+    auto it = mSuppressionTimes.begin ();
```

Comment:
Beyond the scope of the PR but this looks like a candidate for `beast::aged_unordered_map`